mjidris/irondb
# Us168

## Objective
Rebuild Database view and connect the home page to it

## Detailed Description
This is a large update that features the database view, which contains a table that displays meteorite information, and a search bar used to make searches throughout the table. The Database page also features a reset button that clears out the search form and resets the table to display all entries. The user is also able to make searches straight from the home page.

The export function does not work at the moment. The button doesn't do much at the moment.

## Screenshot (if applicable)
Initial View
![PR1](https://user-images.githubusercontent.com/47923772/79702994-a335c180-825d-11ea-9c1a-25ac196e5c90.png)

Expanded search form w/ search made
![PR2](https://user-images.githubusercontent.com/47923772/79702997-a5981b80-825d-11ea-859f-853766657013.png)

Empty search results
![PR3](https://user-images.githubusercontent.com/47923772/79702999-a761df00-825d-11ea-81ea-47cf32b56074.png)

## How to Test (if applicable)

**To spin up branch, stop and delete all current docker volumes/images and rebuild:**
`./iron.sh -s`
`docker volume prune && docker image prune`
`docker image rm irondb_pgdb`
`./iron.sh -i`

**Test home page search into database page**
1. From the home page, enter 'guanaco' in the name field
2. Hit search
3. Observe that the table displays all entries with the name 'guanaco'
4. Make any other searches you would like and observe that the table displays the correct entries

**Test database page searches**
_Note: The search function can handle multiple search criteria_
1. From the database page, enter any number of search criteria into the search form
2. Hit search
3. Observe that the table displays the correct entries

**Test reset button**
_Note: The volume and year fields will only reset if a numerical value is entered. "123avd" will not work_
1. Enter appropriate information into each input/select field
2. Make a random search of any any kind
3. Observe that all search fields are reset to be either empty for input fields, or a default value for the select elements
4. Observe that the table resets to display all entries

 **Test the consistency of the spacing between the search form and the table**
_Note: This one comes from the list of bug fix requests that we got in the beginning_
1. From the database page, hit the 'plus' and 'minus' sign icons expand and contract the search form
2. Observe that the spacing stays consistent with each search element that is removed/added



## Tests (if added)
N/A

## Requirements
[Taiga User Story]
https://tree.taiga.io/project/cbrobsto-iron-db/us/168?milestone=260360
[Taiga Task]
https://tree.taiga.io/project/cbrobsto-iron-db/task/305?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/325?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/325?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/274?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/275?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/273?kanban-status=1694342
https://tree.taiga.io/project/cbrobsto-iron-db/task/280?kanban-status=1694342
****
